File: 1. Two Sum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        int sum = 0;
        vector<int> vec(2);
      for(int i = 0; i < nums.size(); i++)  
        for(int j = 0; j < nums.size(); j++){
            sum = nums.at(i) + nums.at(j);
            if(sum == target && i != j){
                vec.at(0) = i;
                vec.at(1) = j;
                return vec;
            }
           
        } 
        
        return vec;
    }
};

File: 1. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> mapping = new HashMap<Integer,Integer>(); //number to index
	        
		int[] a = new int[2];
	
		for(int i = 0; i < nums.length;i++) {

			int num = nums[i];
			if(mapping.containsKey(target - nums[i])) {
				a[0] = mapping.remove(target - nums[i]);
				a[1] = i;
				return a;
			}else {
				mapping.put(nums[i],i);
			}
		}
		
	
		return a;
        
    }
    
}
